app/src/ui/rebase/choose-branch.tsx

```diff
@@ -160,19 +164,9 @@ export class RebaseBranchDialog extends React.Component<
       return
     }
 
-    // TODO: transition to a rebase progress dialog
-
-    this.setState({ isRebasing: true })
-
-    await this.props.dispatcher.rebase(
-      this.props.repository,
-      branch.name,
-      this.props.currentBranch.name
-    )
-
-    this.setState({ isRebasing: false })
+    // TODO: compute the number of expected commits that will be rebased
 
-    this.props.onDismissed()
+    this.props.onStartRebase(branch.name, this.props.currentBranch.name, 20)
```

Comment:
This magic number `20` is a placeholder because I'm going to compute the right number of commits that will be rebased as part of #6959, and pass that through to the rebase flow.

Reply:
It lives! #7167